### PR TITLE
Kapacitor Flux tasks

### LINKED ISSUE #2564: Flux task support in Kapacitor
https://github.com/influxdata/kapacitor/pull/2555
https://github.com/influxdata/kapacitor/pull/2544

This feature allows users to run flux tasks using Kapacitor. There is a detailed demo video available.

Config example:

```
[fluxtask]
  enabled = true
  task-run-influxdb = ""
  task-run-bucket = "taskbucket"
  task-run-org = ""
  task-run-orgid = ""
  task-run-measurement = "runs"
  [fluxtask.secrets]
    my_token = "a_very_secret_token"
    my_other_token = "a_different_one"
```

Note `task-run-bucket` is required config (it defaults to the empty string, which is invalid if `enabled=true`), and Kapacitor does not create this bucket automatically, it must be created in the associated influxdb instance (`task-run-influxdb` or else the `default` influxdb).

The exact API endpoints are TBD but it will conform to:

https://docs.influxdata.com/influxdb/cloud/api/#tag/Tasks

except that Kapacitor does not have `owners`, `members`, or `labels` related APIs. Kapacitor also does not have a fine grained ACL, so any user that has permission to use the Kapacitor API can view / update / delete all tasks from all users.

### PR DESCRIPTION
**Pull request by sanderson**
Closes #2564 

- Introduces task-based docs for setting up and managing Kapacitor Flux tasks
- Introduces new shortcode for displaying API request methods and endpoints
- Updates the `kapacitor` CLI docs with flux task commands
- Holding off on updating the API doc with Flux task endpoints until @jstirnaman wraps up the users endpoint docs

---

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -333,6 +333,18 @@ current product. Easier to maintain being you update the version number in the `
...
+### API endpoint
+Use the `{{< api-endpoint >}}` shortcode to generate a code block that contains
+a colored request method and a specified API endpoint.
+Provide the following arguments:
+
+- **method**: HTTP request method (get, post, patch, put, or delete)
+- **endpoint**: API endpoint
+
+```md
+{{< api-endpoint method="get" endpoint="/api/v2/tasks">}}
```
Comment (pierwill) [inline CONTRIBUTING.md:345]: 🔥

File: content/kapacitor/v1.6/working/cli_client.md [hunk 2]
```diff
@@ -35,9 +24,17 @@ arguments applicable to that command.
 kapacitor [options] [command] [args]
 ```
 
-This document provides an overview of all the commands provided by the `kapacitor` CLI.
-These include general options, core commands, server management, data sampling,
-working with topics and topic handlers, and working with tasks and task templates.
+This document provides an overview of all the commands provided by the `kapacitor` CLI:
```
Comment (pierwill) [inline content/kapacitor/v1.6/working/cli_client.md:27]:
```suggestion
This document provides an overview of the commands provided by the `kapacitor` CLI:
```

File: content/kapacitor/v1.6/working/cli_client.md [hunk 3]
```diff
@@ -1095,3 +1092,160 @@ the pattern `"generic*"`.
...
+- [`flux task run list`](#flux-task-run-list)
+- [`flux task run retry`](#flux-task-run-retry)
+
+---
+
+### flux task create
+The `kapacitor flux task create` command creates a new Kapacitor Flux task.
+
+```sh
+kapacitor flux task create [flags] [flux script or '-' for stdin]
```
Comment (pierwill) [inline content/kapacitor/v1.6/working/cli_client.md:1113]:
```suggestion
kapacitor flux task create [flags] [flux script or '-' to read from stdin]
```

File: content/kapacitor/v1.6/working/flux/_index.md [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+  - /influxdb/cloud/process-data/common-tasks/
+  - /influxdb/cloud/process-data/task-options/
+---
+
+Use **Kapacitor 1.6+** to run Flux tasks against InfluxDB and other data sources.
+Leverage the full library of Flux functionality to build powerful data processing
+and monitoring tasks in Kapacitor.
+
+## Before you start
+Before you get started, there are things to consider when using Kapacitor Flux tasks:
```
Comment (pierwill) [inline content/kapacitor/v1.6/working/flux/_index.md:25]:
```suggestion
Before you get started with Flux tasks, consider:
```

File: content/kapacitor/v1.6/working/flux/_index.md [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+
+- Kapacitor Flux tasks do not use Kapacitor topics or event handlers.
```
Comment (pierwill) [inline content/kapacitor/v1.6/working/flux/_index.md:27]: Do not or _cannot_?

File: content/kapacitor/v1.6/working/flux/_index.md [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+_For more details about creating Kapacitor Flux tasks, see [Create a Kapacitor Flux task](/kapacitor/v1.6/working/flux/manage/create/)._
+{{% /tab-content %}}
+<!-------------------------- END InfluxDB 1.x content ------------------------->
+<!---------------------- BEGIN InfluxDB Cloud/2.x content --------------------->
+{{% tab-content %}}
+{{% warn %}}
+#### Consider using InfluxDB tasks
+If you're using **InfluxDB Cloud** or **InfluxDB OSS 2.x**, you certainly can
+use Kapacitor to run Flux tasks against them, but you should consider just using
+[native InfluxDB tasks](/influxdb/cloud/process-data/).
```
Comment (pierwill) [inline content/kapacitor/v1.6/working/flux/_index.md:182]:
```suggestion
If you're using **InfluxDB Cloud** or **InfluxDB OSS 2.x**, consider using
[native InfluxDB tasks](/influxdb/cloud/process-data/) for data processing.
```

File: content/kapacitor/v1.6/working/flux/manage/update.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+{{< tabs-wrapper >}}
+{{% tabs %}}
+[CLI](#)
+[API](#)
+{{% /tabs %}}
+<!----------------------------- BEGIN CLI content ----------------------------->
+{{% tab-content %}}
+
+Use the `kapacitor flux task update` command to update Kapacitor Flux task.
```
Comment (pierwill) [inline content/kapacitor/v1.6/working/flux/manage/update.md:22]:
```suggestion
Use the `kapacitor flux task update` command to update Kapacitor Flux tasks.
```

**Review by pierwill — APPROVED** [review]

File: content/kapacitor/v1.6/working/flux/_index.md (hunk 5 above)
Comment (sanderson) [reply content/kapacitor/v1.6/working/flux/_index.md:27]: Umm both 😄

**Pull request merged**